dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/trackedentity/DefaultTrackedEntityService.java

```diff
@@ -253,35 +234,136 @@ public TrackedEntity getTrackedEntity(
+    TrackedEntity trackedEntity = getTrackedEntity(uid);
+    UserDetails currentUser = getCurrentUserDetails();
+
     List<String> errors = trackerAccessManager.canRead(currentUser, trackedEntity);
+    if (!errors.isEmpty()) {
+      throw new ForbiddenException(errors.toString());
+    }
 
+    return mapTrackedEntity(trackedEntity, params, currentUser, includeDeleted);
+  }
+
+  /**
+   * Gets a tracked entity based on the program and org unit ownership
+   *
+   * @return the TE object if found and accessible by the current user
+   * @throws NotFoundException if uid does not exist
+   * @throws ForbiddenException if TE owner is not in user's scope or not enough sharing access
+   */
+  private TrackedEntity getTrackedEntity(
+      String uid, Program program, TrackedEntityParams params, boolean includeDeleted)
+      throws NotFoundException, ForbiddenException {
+    TrackedEntity trackedEntity = trackedEntityStore.getByUid(uid);
+    addTrackedEntityAudit(trackedEntity, CurrentUserUtil.getCurrentUsername());
+    if (trackedEntity == null) {
+      throw new NotFoundException(TrackedEntity.class, uid);
+    }
+
+    UserDetails userDetails = getCurrentUserDetails();
+    List<String> errors =
+        trackerAccessManager.canReadProgramAndTrackedEntityType(
+            userDetails, trackedEntity, program);
     if (!errors.isEmpty()) {
       throw new ForbiddenException(errors.toString());
     }
 
+    String error =
+        trackerAccessManager.canAccessProgramOwner(userDetails, trackedEntity, program, false);
+    if (error != null) {
+      throw new ForbiddenException(error);
+    }
+
+    return mapTrackedEntity(trackedEntity, params, userDetails, includeDeleted);
+  }
+
+  /**
+   * Gets the requested tracked entity if the user owns at least one TE/program pair, or has access
+   * to the TE registering org unit, in case it doesn't own any.
+   *
+   * @return the TE object if found and accessible by the user
+   * @throws NotFoundException if TE does not exist
+   * @throws ForbiddenException if TE is not accessible
+   */
+  private TrackedEntity getTrackedEntity(String uid) throws NotFoundException, ForbiddenException {
+    TrackedEntity trackedEntity = trackedEntityStore.getByUid(uid);
+    addTrackedEntityAudit(trackedEntity, CurrentUserUtil.getCurrentUsername());
+    if (trackedEntity == null) {
+      throw new NotFoundException(TrackedEntity.class, uid);
+    }
+
+    initializeTrackedEntityOrgUnitParents(trackedEntity);
+
+    if (programService.getAllPrograms().stream()
+        .anyMatch(
+            p ->
+                p.getTrackedEntityType() == trackedEntity.getTrackedEntityType()
+                    && trackerAccessManager
+                        .canRead(getCurrentUserDetails(), trackedEntity, p, false)
+                        .isEmpty())) {
+      return trackedEntity;
+    }
+
+    throw new ForbiddenException(TrackedEntity.class, uid);
+  }
+
+  /**
```

Comment:
It's a purely style comment. I would put this java doc in the call of the method so it is even more clear that the workaround is the call to the method itself